### DISCUSSION TITLE
Why does this library need to care about types returned by RPC methods?

### DISCUSSION THREAD

**Discussion by keysmashes** [permitio/fastapi_websocket_rpc#47] [General]
If you don't put a type annotation on RPC methods, the returned value gets converted to a string:

https://github.com/permitio/fastapi_websocket_rpc/blob/307c14980663ad18cc4f4a73eb98b858a87704d2/fastapi_websocket_rpc/rpc_channel.py#L344-L346

I don't understand why this is; when serialisation occurs, it's just by asking Pydantic to do whatever it thinks is best:

https://github.com/permitio/fastapi_websocket_rpc/blob/307c14980663ad18cc4f4a73eb98b858a87704d2/fastapi_websocket_rpc/utils.py#L77

and deserialisation is even simpler:

https://github.com/permitio/fastapi_websocket_rpc/blob/307c14980663ad18cc4f4a73eb98b858a87704d2/fastapi_websocket_rpc/simplewebsocket.py#L41

so I don't see why fastapi_websocket_rpc needs to care about the types in use at all. Am I missing something?

My specific use-case is that I was trying to return a list from an RPC call, and was confused why it was ending up as a string on the other side. Annotating the method with `-> typing.Any` was enough to stop the automatic conversion to `str`.

**Comment by orweis** [comment]
Hi Thanks for reaching out.

You're basically asking - "why use types at all ?". ;-) 
Typing is a defence mechanism that helps us developers make sure we connect interfaces correctly. In this case it helps you check your RPC clients and servers are aligned on expectations - this would work well together with tests you can potentially write.
Here's [a decent post on the concept for further reading ](https://medium.com/@moraneus/exploring-the-power-of-pythons-typing-library-ff32cec44981)
Fast-API unlike some other frameworks in Python has really come to fame thanks to its use of typing and Pydantic, so I'd say this would be the natural affinity of this project.

As you have a workaround for you need (though I'd encourage you to use better typing to improve the robustness of your code) this is not an issue - I'm converting this to a discussion.  <3 

Hope you find this information useful. :)

**Reply by keysmashes** [reply]
I am familiar with the concept :) I'm more asking "why does fastapi_websocket_rpc do implicit conversion to string"? Implicit type coercion leads to many footguns; for example the first example in the README:

```python
class RpcCalculator(RpcMethodsBase):
    async def add(self, a, b):
        return a + b
```

This lacks a type annotation and so it will return a string when called via RPC, not an int or a float as you might expect.
